src/autorest-core/lib/message.ts

```diff
@@ -30,7 +30,10 @@ export interface Message {
   Key?: Iterable<string>;
   Details?: any;
   Text: string;
+
+  // injected or modified by core
   Source?: Array<SourceLocation>;
   Range?: Iterable<Range>;
   Plugin?: string;
+  FormattedMessage?: string;
```

Comment:
@olydis I agree, I think it'd be applicable to all messages, so I'd vote for adding it. Yes, the formatted message should be evaluated every time something gets added, in this case, it probably should be added either to formatted message or details or both, so it's printed in CI/command line with the rest.